### PR TITLE
feat(license): add struct with types for licenses 

### LINKED ISSUE #5204: don't split licenses from `License` field from python packaging
### Discussed in https://github.com/aquasecurity/trivy/discussions/5155

`License` field in `.dist-info/METADATA` files contains textual information about license/exceptions/specific versions/etc... - https://packaging.python.org/en/latest/specifications/core-metadata/#license

We can't split this text correctly.
Save this field without division.

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add new `License` struct to exclude prefixes for licenses.

TODO:
- add filePath for licenses (https://github.com/aquasecurity/trivy/discussions/6117#discussioncomment-8463597)

## Related issues
- Close #5204

## Related PRs
- [ ] aquasecurity/go-dep-parser/pull/256

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by lior-orca** [comment]
@DmitriyLewen is this PR stale or do you intend to continue with it in the near future?

**Comment by DmitriyLewen** [comment]
Hello @lior-orca 
This PR is ready, but https://github.com/aquasecurity/go-dep-parser/pull/256 needs to be merged first.

**Comment by dus7eh** [comment]
Please check my comments in #5204

File: pkg/fanal/types/license.go [hunk 1]
```diff
@@ -56,3 +65,62 @@ type LicenseFinding struct {
...
+	for _, l := range stringLicenses {
+		licenses = append(licenses, License{
+			Type:  LicenseTypeName,
+			Value: l,
+		})
+	}
+
+	*ll = licenses
+	return nil
+}
```
Comment (DmitriyLewen) [inline pkg/fanal/types/license.go:126]: Looks like we don't need to save license types for `convert` mode.

**Comment by Rafcu90** [comment]
Hello @DmitriyLewen I would like to ask is there any estimation when this can be delivered ? My team is strongly interested in ignoring python packages by their paths as it was discussed under: https://github.com/aquasecurity/trivy/discussions/6117. Thank you in advance, Regards Rafał :)

**Comment by DmitriyLewen** [comment]
Hello @Rafcu90 
Unfortunately, I can't give a time frame.

This PR changes base structure (I mean `Package`). The necessary time to review and test these changes.
We've also combined `go-dep-parser` with Trivy. I need time to move https://github.com/aquasecurity/go-dep-parser/pull/256 to this PR.

**Pull request closed without merging**